monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -3125,6 +3125,27 @@ abstract class Observable[+A] extends Serializable { self =>
   final def takeUntil(trigger: Observable[Any]): Observable[A] =
     new TakeUntilObservable[A](self, trigger)
 
+  /** Version of [[takeUntil]] that can work with a trigger expressed by a [[monix.eval.Task]]
+    *
+    * @see [[takeUntil]] for version that works with Observable.
+    * @see [[takeUntilEvalF]] for version that works with generic `F[_]` powered by [[monix.eval.TaskLike]].
+    *
+    * @param trigger task that will cancel the stream as soon as it completes.
+    */
+  final def takeUntilEval(trigger: Task[Any]): Observable[A] =
```

Comment:
Instead of `Task[Any]` we could use `Task[_]`. Same for the `Observable` and the `F[_]` versions.

The `F[_]` version isn't even correct because it isn't specified as a covariant type and so it doesn't really work.

So change `Any` to `_`.

Reply:
Ah yes, you're right. I've updated the PR.